test/apps/iodemo/ucx_wrapper.h

```diff
@@ -86,6 +87,18 @@ class UcxContext {
         UcxConnection &_connection;
     };
 
+    class UcxDisconnector : public UcxCallback {
```

Comment:
can we remove this class, and let io_demo layer inherit from UcxCallback for its disconnect callbacks?

Reply:
this class is used in `~UcxContext`, we can make this class as a base for io_demo layer but see https://github.com/openucx/ucx/pull/6548#discussion_r605498320 conversation

Reply:
i think io_demo layer should use UcxCallback as base for disconnect callbacks

Reply:
> i think io_demo layer should use UcxCallback as base for disconnect callbacks

this is how it's implemented now

Reply:
1. make it private
2. maybe remove UcxDisconnector, and simplify io_demo.cc, by defining that disconnect() always destroys the connection after calling the callback (from progress_disconnected_connections, if disconnect_progress returned true)?

Reply:
> maybe remove UcxDisconnector, and simplify io_demo.cc, by defining that disconnect() always destroys the connection after calling the callback (from progress_disconnected_connections, if disconnect_progress returned true)?

we cannot add `UcxConnection* UcxContext::connect` function because of code on client side:
```
        UcxConnection *conn = new UcxConnection(*this, opts().use_am);
        _server_info[server_index].conn = conn;
        conn->connect((const struct sockaddr*)&connect_addr,
                      sizeof(connect_addr),
                      new ConnectCallback(*this, server_index));
```
which links `conn`, `_server_info` and `new ConnectCallback(*this, server_index)`, so IMO better to have `new`/`delete` ownership on the same level